### PR TITLE
Add a method to publish a list of repos to a prefix

### PR DESCRIPTION
**Pull request by shadeslayer**
Allow publishing a list of repositories with different components
to a single prefix

### CODE REVIEW

File: lib/aptly/publish.rb [hunk 1]
```diff
@@ -42,6 +42,16 @@ def list(connection = Connection.new, **kwords)
                                    query: kwords)
         JSON.parse(response.body).collect { |h| new(connection, h) }
       end
+
+      # Publish a [Array<PublishedRepository>] list of repositories to a prefix
```
Comment (hsitter) [inline lib/aptly/publish.rb]: `Array<Repository>`

File: test/integration.rb [hunk 2]
```diff
@@ -99,10 +99,28 @@ def test_eee_repo_publish
...
+    kittenRepo.edit!(DefaultComponent: 'kitten')
+    puppyRepo = ::Aptly::Repository.create('puppy', DefaultComponent: 'puppy')
+    pub = Aptly::PublishedRepository.from_repositories([kittenRepo, puppyRepo], 'kewl-repo-name', Distribution: 'wily', Architectures: %w[amd64], Signing: { Skip: true })
+
+    assert pub.is_a? ::Aptly::PublishedRepository
+    assert_equal 'wily', pub.Distribution
+    assert_equal %w[amd64], pub.Architectures
+    assert_equal %w[kitten puppy] , pub.Sources.collect(&:Name).sort
+    assert pub.Sources[0].is_a? ::Aptly::Repository
+
```
Comment (hsitter) [inline test/integration.rb]: Might be worth it to assert the final component. Same for the unit test.

**Review by hsitter — APPROVED** [review]
Some nitpicks. Otherwise LGTM

**Comment by shadeslayer** [comment]
Fixed nitpicks

**Pull request merged**